lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,58 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        self.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        self.memory_query_data = memory_parse_data(memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        self.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { memory_query_data }
+      end
+
+      def memory_short_form_data
+        proc { memory_query_data }
+      end
+
+      def memory_query_data
+        Thread.current[:miq_perf_request_memory_data]
+      end
+
+      def memory_query_data=(value)
+        Thread.current[:miq_perf_request_memory_data] = value
+      end
+
+      def include_memsize?
```

Comment:
Sorry, I should have mentioned this in the last review, can we namespace this method as well?